### ISSUE TITLE
Issues related to Audio & HTML output

### ISSUE THREAD

**Issue by abidlabs** [gradio-app/gradio#519]
**Describe the bug**

**Issue 1. I had issues outputting the audio.**
I realized there was no example showing audio output with numpy array + sampling rate. All examples save to a file and return file path. The docs are a bit confusing saying it expects the sample rate + " the data numpy.array of shape (samples, 2)".
I returned this but it gave me no audio (didn't fail, just showed no audio)
I also tried saving the audio to a wav file. I was able to play locally after downloading and also play in colab. So I tried to return that file, and I got bad sample width error.
I dived into the gradio source code. I pretty much replicated audio_to_file which is what's used in postprocessing but with the changes from this stack overflow question https://stackoverflow.com/questions/54261690/python-pydub-bad-sample-width-error-when-exporting-audiosegment-created-from-nu. With this I was able to return an audio (yay) but it was sounding very bad/different than the original wav I was saving directly with scipy.io.wavfile.write.

**Issue 2. Returning the HTML is not working in this demo**
I'm converting bokeh output to HTML, but this does not seem to be working.
The HTML Output is in format of full file, so maybe that's the issue. Unfortunately I don't see any logs that could help figure out the issue. Some documentation on limitations of HTML output could be useful

```html
<!DOCTYPE html>
<html lang="en">
...
```

**To Reproduce**
https://huggingface.co/spaces/osanseviero/gpt2_for_music
https://colab.research.google.com/drive/1fWi-x9HjxM2FiHnXK7hq34qhVbPZA0jy

**Comment by abidlabs** [comment]
**Issue 1**:

We actually have two examples showing how to return Audio with a numpy array and a sampling rate. I just tested both of them and they seem to work fine:

- https://colab.research.google.com/drive/1QDUedWIobKp3VOfmsxjHwMREWyMGz5gb
- https://colab.research.google.com/drive/1Ol-Z4x6j5auCRXxJflMX6EDBYx5oFRe0#scrollTo=e200MmBU2aLT

Does this address your problem?

**Issue 2**:

Again, I don't see this issue. For example, you can try this code:

```python
import gradio as gr

def test(x):
    x = """
<!DOCTYPE html>
<html>
<body>

<h1>My First Heading</h1>
<p>My first paragraph.</p>

</body>
</html>    
"""
    return x

gr.Interface(test, "text" "html").launch()
```
This works for me. 

cc @osanseviero

**Comment by osanseviero** [comment]
Thanks for investigating! I was able to fix the first issue :rocket: 

TL;DR
`gradio` expects integer array (as documented) but will fail when working with float audio files.

Here are my notes

**Issue 1. Audio output issues**

Thanks for the examples! Where are they? I was not able to find them in https://gradio.app/ nor https://github.com/gradio-app/awesome-demos/blob/main/README.md

1. What I tried first: return `np.array` and sample rate ([colab](https://colab.research.google.com/drive/1N0GRkFsqVlba7n8hMVrQC_PrgzNTNIq_?usp=sharing))

I compared with your examples and it seems like `note_seq` synthethizer returns a 1-D numpy float array containing the synthesized waveform. while the expectation is to use integers. (This means `gradio` does not work with floating point audios although they are valid)

2. Just to confirm, I tried something different: save the audio to a file and try to output it ([colab](https://colab.research.google.com/drive/1bLOOoQZiINOQ8ziq_cZdF_X6iBjVyeww#scrollTo=0UBs1_4OMMbP))

The only thing I added was

```python
scipy.io.wavfile.write("test_file.wav", SAMPLE_RATE, array_of_floats)
```

and now the interface outputs `"test_file.wav"`

Note: I'm able to download the `wav` file and play it, meaning the wav works well. Instead, the interface shows an unplayable output.

3. Float to int. This worked!

The final solution was to convert the floats to int. Just something like this worked to me (based on [this](https://stackoverflow.com/questions/48144381/changing-wav-format-from-32-bit-floating-point-to-16-bit-pcm))

```python
array_of_floats = synth(note_sequence, sample_rate=SAMPLE_RATE)
array_of_floats /=1.414
array_of_floats *= 32767
int16_data = array_of_floats.astype(np.int16)
```

**Comment by abidlabs** [comment]
> Thanks for the examples! Where are they? I was not able to find them in https://gradio.app/ nor https://github.com/gradio-app/awesome-demos/blob/main/README.md

They are in the docs for the Audio Output component here: https://gradio.app/docs/#o_audio

> The final solution was to convert the floats to int. Just something like [this](https://stackoverflow.com/questions/48144381/changing-wav-format-from-32-bit-floating-point-to-16-bit-pcm) worked to me (based on this)

I'll implement this change in the library, thanks so much for reporting this.

Let me know if you have any updates on the 2nd issue!

**Comment by abidlabs** [comment]
@osanseviero let us know if you'd like to keep this issue open. I'm inclined to close it since the Audio issue is solved, and the 2nd issue seems pretty convoluted. If you have more info about the 2nd issue, let us know!

**Comment by osanseviero** [comment]
Hey @abidlabs! I think we can close this issue, as the 2nd problem is an ongoing issue with Bokeh generated HTML and does not seem to be for Gradio
